internal/cli/atlas/serverless/backup/restores/list.go

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 MongoDB Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package restores
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli"
+	"github.com/mongodb/mongodb-atlas-cli/internal/cli/require"
+	"github.com/mongodb/mongodb-atlas-cli/internal/config"
+	"github.com/mongodb/mongodb-atlas-cli/internal/flag"
+	"github.com/mongodb/mongodb-atlas-cli/internal/store"
+	"github.com/mongodb/mongodb-atlas-cli/internal/usage"
+	"github.com/spf13/cobra"
+)
+
+type ListOpts struct {
+	cli.GlobalOpts
+	cli.OutputOpts
+	cli.ListOpts
+	clusterName string
+	store       store.ServerlessRestoreJobsLister
+}
+
+func (opts *ListOpts) initStore(ctx context.Context) func() error {
+	return func() error {
+		var err error
+		opts.store, err = store.New(store.AuthenticatedPreset(config.Default()), store.WithContext(ctx))
+		return err
+	}
+}
+
+var restoreListTemplate = `ID	SNAPSHOT	CLUSTER	TYPE	EXPIRES AT{{range .Results}}
+{{.ID}}	{{.SnapshotID}}	{{.TargetClusterName}}	{{.DeliveryType}}	{{.ExpiresAt}}{{end}}
+`
+
+func (opts *ListOpts) Run() error {
+	listOpts := opts.NewListOptions()
+	r, err := opts.store.ServerlessRestoreJobs(opts.ConfigProjectID(), opts.clusterName, listOpts)
+	if err != nil {
+		return err
+	}
+
+	return opts.Print(r)
+}
+
+// mongocli atlas backup(s) restore(s) job(s) list <clusterName> [--page N] [--limit N].
+func ListBuilder() *cobra.Command {
+	opts := new(ListOpts)
+	cmd := &cobra.Command{
+		Use:     "list <clusterName>",
+		Aliases: []string{"ls"},
+		Short:   "Return all cloud backup restore jobs for your project and cluster.",
+		Long:    fmt.Sprintf(usage.RequiredRole, "Project Owner"),
+		Args:    require.ExactArgs(1),
+		Annotations: map[string]string{
+			"clusterNameDesc": "Name of the Atlas cluster for which you want to retrieve restore jobs.",
+		},
+		Example: fmt.Sprintf(`  # Return all continuous backup restore jobs for the cluster Cluster0:
+  %s serverless backup restore list Cluster0`, cli.ExampleAtlasEntryPoint()),
```

Comment:
Also changing the example name to avoid confusion
```suggestion
		Example: fmt.Sprintf(`  # Return all continuous backup restore jobs for the serverless instance Instance0:
  %s serverless backup restore list Instance0`, cli.ExampleAtlasEntryPoint()),
```